### PR TITLE
deposit: add missing relation types

### PR DESCRIPTION
**Pull request by ctrl-alt-soneca**

### CODE REVIEW

**Review by jbenito3 — APPROVED** [review]

File: zenodo/modules/records/config.py [hunk 1]
```diff
@@ -99,6 +101,10 @@
     ('isCompiledBy', _('Compiled by')),
     ('isDerivedFrom', _('Derived from')),
     ('isSourceOf', _('Source of')),
+    ('requires', _('Requires')),
+    ('isRequiredBy', _('Required material')),
```
Comment (slint) [inline zenodo/modules/records/config.py:105]:
```suggestion
    ('isRequiredBy', _('Required by')),
```

Just for consistency with the rest of the terms I guess

File: zenodo/modules/records/data/include/v4.1/datacite-relationType-v4.1.xsd [hunk 2]
```diff
@@ -16,6 +16,8 @@
       <xs:enumeration value="IsSupplementedBy"/>
       <xs:enumeration value="IsContinuedBy"/>
       <xs:enumeration value="Continues"/>
+      <xs:enumeration value="IsDescribedBy"/>
```
Comment (slint) [inline zenodo/modules/records/data/include/v4.1/datacite-relationType-v4.1.xsd:19]: This file we don't need to modify... We used to validate at some point our DataCite export, but actually don't do that anymore.

I'll create an issue from here to actually remove these files completely.

File: zenodo/modules/deposit/static/json/zenodo_deposit/deposit_form.json [hunk 3]
```diff
@@ -719,6 +727,22 @@
...
+                "value": "isRequiredBy",
+                "name": "requires this upload"
+              },
+              {
+                "value": "isObsoletedBy",
+                "name": "replaces this upload"
+              },
+              {
+                "value": "obsoletes",
+                "name": "replaced by this upload"
```
Comment (slint) [inline zenodo/modules/deposit/static/json/zenodo_deposit/deposit_form.json:744]:
```suggestion
                "name": "is replaced by this upload"
```

**Review by slint — APPROVED** [review]
Some minor comments for consistency, otherwise LGTM :+1:

File: zenodo/modules/records/data/include/v4.1/datacite-relationType-v4.1.xsd (hunk 2 above)
Comment (ctrl-alt-soneca) [reply zenodo/modules/records/data/include/v4.1/datacite-relationType-v4.1.xsd:19]: Ok! Wil discard theses changes then 👍

**Comment by slint** [comment]
@ctrl-alt-soneca Last thing, can you check if there's a relevant issue to be closed?

**Pull request merged**